### ISSUE TITLE
SMplayer not recognized as a player

### ISSUE THREAD

**Issue by nikgtasa** [Syncplay/syncplay#584]
**Describe the bug**
Installed SMPlayer and tried hooking it up to Syncplay but selecting the smplayer.exe and starting syncplay loops back to the config window with message "File must be selected before starting your player". If i do select a video then the regular room interface opens but without the player. If i try to input videos into the playlist and select them to play nothing happens either, only an error "Could not find file ****.mkv in media directories for playlist switch!". I could choose the mpv.exe and it works but what's the point of that if i wanted specifically a vlc-like interface for mpv.

**To Reproduce**
Install SMPlayer
Select path to player program files\smplayer\smplayer.exe
Run config

**Version and platform:**
 - OS: WIN10 64 22H2
 - Syncplay version and build type: 1.7.0 beta 1 portable
 - Media player and version: SMPlayer Version: 22.7.0 (revision 10091) (64-bit)
Using Qt 5.15.2 (compiled with Qt 5.15.2)
Using MPV 0.34.0

After closing the room window i get a notification about errors in the log file. Here's the ones from .syncplay.log in appdata/roaming:
Traceback (most recent call last):
  File "C:\projects\syncplay\syncplay\ui\gui.py", line 1745, in changeReadyState
  File "C:\projects\syncplay\syncplay\client.py", line 868, in wrapper
  File "C:\projects\syncplay\syncplay\client.py", line 980, in changeReadyState
  File "C:\projects\syncplay\syncplay\client.py", line 868, in wrapper
  File "C:\projects\syncplay\syncplay\client.py", line 974, in toggleReady
AttributeError: 'NoneType' object has no attribute 'setReady'
Traceback (most recent call last):
  File "C:\projects\syncplay\syncplay\ui\gui.py", line 1745, in changeReadyState
  File "C:\projects\syncplay\syncplay\client.py", line 868, in wrapper
  File "C:\projects\syncplay\syncplay\client.py", line 980, in changeReadyState
  File "C:\projects\syncplay\syncplay\client.py", line 868, in wrapper
  File "C:\projects\syncplay\syncplay\client.py", line 974, in toggleReady
AttributeError: 'NoneType' object has no attribute 'setReady'
Traceback (most recent call last):
  File "C:\projects\syncplay\syncplay\ui\gui.py", line 1745, in changeReadyState
  File "C:\projects\syncplay\syncplay\client.py", line 868, in wrapper
  File "C:\projects\syncplay\syncplay\client.py", line 980, in changeReadyState
  File "C:\projects\syncplay\syncplay\client.py", line 868, in wrapper
  File "C:\projects\syncplay\syncplay\client.py", line 974, in toggleReady
AttributeError: 'NoneType' object has no attribute 'setReady'
Traceback (most recent call last):
  File "syncplay\ui\gui.pyc", line 463, in wrapper
  File "syncplay\ui\gui.pyc", line 1027, in browseMediapath
AttributeError: 'NoneType' object has no attribute 'customOpenDialog'
Traceback (most recent call last):
  File "syncplay\ui\gui.pyc", line 463, in wrapper
  File "syncplay\ui\gui.pyc", line 1027, in browseMediapath
AttributeError: 'NoneType' object has no attribute 'customOpenDialog'
Traceback (most recent call last):
  File "syncplay\ui\gui.pyc", line 1958, in dropEvent
  File "syncplay\client.pyc", line 598, in openFile
AttributeError: 'NoneType' object has no attribute 'openFile'
Traceback (most recent call last):
  File "syncplay\ui\gui.pyc", line 463, in wrapper
  File "syncplay\ui\gui.pyc", line 1044, in browseMediapath
AttributeError: 'NoneType' object has no attribute 'customOpenDialog'
Traceback (most recent call last):
  File "syncplay\ui\gui.pyc", line 1958, in dropEvent
  File "syncplay\client.pyc", line 598, in openFile
AttributeError: 'NoneType' object has no attribute 'openFile'

And here's the ones from syncplay.log in the directory (why there's two of them update at the same time i wouldn't know):
Unhandled Error
Traceback (most recent call last):
  File "twisted\python\context.pyc", line 117, in callWithContext
    
  File "twisted\python\context.pyc", line 82, in callWithContext
    
  File "syncplay\vendor\qt5reactor.pyc", line 164, in _read
    
  File "syncplay\vendor\qt5reactor.pyc", line 270, in _iterate
    
--- <exception caught here> ---
  File "twisted\internet\base.pyc", line 991, in runUntilCurrent
    
  File "syncplay\players\mplayer.pyc", line 243, in run
    
  File "syncplay\players\mplayer.pyc", line 46, in __init__
    
  File "syncplay\players\mplayer.pyc", line 306, in drop
    
builtins.AttributeError: 'MplayerPlayer' object has no attribute '_listener'

Unhandled Error
Traceback (most recent call last):
  File "syncplay\client.pyc", line 878, in start
    
  File "syncplay\vendor\qt5reactor.pyc", line 303, in run
    
  File "syncplay\vendor\qt5reactor.pyc", line 276, in iterate_qt
    
  File "syncplay\vendor\qt5reactor.pyc", line 363, in iterate
    
--- <exception caught here> ---
  File "twisted\internet\base.pyc", line 991, in runUntilCurrent
    
  File "syncplay\players\mplayer.pyc", line 243, in run
    
  File "syncplay\players\mplayer.pyc", line 46, in __init__
    
  File "syncplay\players\mplayer.pyc", line 306, in drop
    
builtins.AttributeError: 'MplayerPlayer' object has no attribute '_listener'

Unhandled Error
Traceback (most recent call last):
  File "twisted\internet\defer.pyc", line 477, in addCallbacks
    
  File "twisted\internet\defer.pyc", line 857, in _runCallbacks
    
  File "twisted\internet\base.pyc", line 510, in _continueFiring
    
  File "twisted\internet\base.pyc", line 762, in disconnectAll
    
--- <exception caught here> ---
  File "twisted\python\log.pyc", line 96, in callWithLogger
    
  File "twisted\python\log.pyc", line 80, in callWithContext
    
  File "twisted\python\context.pyc", line 117, in callWithContext
    
  File "twisted\python\context.pyc", line 82, in callWithContext
    
  File "twisted\internet\tcp.pyc", line 508, in connectionLost
    
  File "twisted\internet\tcp.pyc", line 326, in connectionLost
    
  File "twisted\protocols\tls.pyc", line 400, in connectionLost
    
  File "twisted\protocols\policies.pyc", line 114, in connectionLost
    
  File "twisted\internet\endpoints.pyc", line 159, in connectionLost
    
  File "twisted\application\internet.pyc", line 466, in connectionLost
    
  File "twisted\application\internet.pyc", line 659, in <lambda>
    
  File "automat\_methodical.pyc", line 244, in _clientDisconnected
    
  File "automat\_methodical.pyc", line 283, in __call__
    
  File "twisted\application\internet.pyc", line 779, in _wait
    
  File "twisted\application\internet.pyc", line 790, in _doWait
    
  File "syncplay\client.pyc", line 855, in retry
    
  File "syncplay\client.pyc", line 1634, in showMessage
    
  File "syncplay\client.pyc", line 1667, in showOSDMessage
    
  File "syncplay\players\mpv.pyc", line 133, in displayMessage
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 600, in wrapper
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 633, in command
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 330, in command
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 89, in send
    
builtins.BrokenPipeError: handle is closed

Unhandled error in Deferred:

Traceback (most recent call last):
  File "syncplay\vendor\qt5reactor.pyc", line 276, in iterate_qt
    
  File "syncplay\vendor\qt5reactor.pyc", line 363, in iterate
    
  File "twisted\internet\base.pyc", line 991, in runUntilCurrent
    
  File "twisted\internet\task.pyc", line 251, in __call__
    
--- <exception caught here> ---
  File "twisted\internet\defer.pyc", line 190, in maybeDeferred
    
  File "syncplay\client.pyc", line 1217, in __displayMessageOnOSD
    
  File "syncplay\client.pyc", line 1130, in <lambda>
    
  File "syncplay\client.pyc", line 1162, in _checkIfYouReAloneInTheRoom
    
  File "syncplay\client.pyc", line 1667, in showOSDMessage
    
  File "syncplay\players\mpv.pyc", line 133, in displayMessage
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 600, in wrapper
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 633, in command
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 330, in command
    
  File "syncplay\vendor\python_mpv_jsonipc\python_mpv_jsonipc.pyc", line 89, in send
    
builtins.BrokenPipeError: handle is closed

Unhandled Error
Traceback (most recent call last):
  File "syncplay\client.pyc", line 878, in start
    
  File "syncplay\vendor\qt5reactor.pyc", line 303, in run
    
  File "syncplay\vendor\qt5reactor.pyc", line 276, in iterate_qt
    
  File "syncplay\vendor\qt5reactor.pyc", line 363, in iterate
    
--- <exception caught here> ---
  File "twisted\internet\base.pyc", line 991, in runUntilCurrent
    
  File "syncplay\players\mplayer.pyc", line 243, in run
    
  File "syncplay\players\mplayer.pyc", line 46, in __init__
    
  File "syncplay\players\mplayer.pyc", line 306, in drop
    
builtins.AttributeError: 'MplayerPlayer' object has no attribute '_listener'

**Comment by Et0h** [comment]
As per https://syncplay.pl/about/ideas/ Syncplay does not support SMplayer. Please use mpv or mpv.net.

**Comment by Et0h** [comment]
Thanks to https://github.com/Syncplay/syncplay/commit/0307cda8506935d1ea346ccad4a668bec1e13e97 Syncplay will no longer incorrectly detect SMPlayer as a mplayer2, and now it will correctly treat it as a non-supported player. Thanks for letting us know of this issue.
